wallet-next/src/service.rs

```diff
@@ -74,7 +88,20 @@ impl WalletService {
     }
 
     async fn check_worker(&self) -> Result<(), tonic::Status> {
+        // If the shared error slot is set, then an error has occurred in the worker
+        // that we should bubble up.
+        if self.error_slot.lock().unwrap().is_some() {
+            return Err(tonic::Status::new(
+                tonic::Code::Internal,
+                format!(
+                    "Worker failed: {}",
+                    self.error_slot.lock().unwrap().as_ref().unwrap()
+                ),
+            ));
+        }
+
         // TODO: check whether the worker is still alive, else fail, when we have a way to do that
```

Comment:
I think this TODO is closed out by these changes(?)

Reply:
I think there's one case that's not covered: if the worker is to crash without setting the `error_slot`, I don't think it would cause the service to die.